pkg/arvo/neo/cod/std/src/fil/messenger.md

```diff
@@ -0,0 +1,557 @@
+++  form
+  ^-  form:neo
+  |_  [=bowl:neo =aeon:neo state=pail:neo]
+  ++  init
+    |=  old=(unit pail:neo)
+    ^-  (quip card:neo pail:neo)
+    [~ sig/!>(~)]
+  ::
+  ++  poke
+    |=  [=stud:neo vax=vase]
+    ^-  (quip card:neo pail:neo)
+    ?+    stud  !!
+        %dm-diff
+      =/  poke  !<(dm-diff vax)
+      ?>  =(%invited -.poke)
+      :_  state
+      :~  :-  (welp here.bowl ~[%dms p/ship.src.bowl])
+          [%make %dm `dm-diff/vax ~]
+      ==
+    ::
+        %groupchat-diff
+      =/  poke  !<(groupchat-diff vax)
+      ?+    -.poke  !!
+          %invited
+        :_  state
+        :~  :-  (welp here.bowl ~[%groupchats p/ship.src.bowl (rear host.poke)])
+            [%make %groupchat `groupchat-diff/vax ~]
+        ==
+      ==
+    ::
+        %messenger-diff
+      ?>  =(our ship.src):bowl
+      =/  poke  !<(messenger-diff vax)
+      ?-    -.poke
+          %new-dm
+        =/  provider  ~[p/partner.poke %home %messenger]
+        :_  state
+        :~  :-  (welp here.bowl ~[%dms p/partner.poke])
+            [%make %dm `dm-diff/!>([%initiate partner.poke provider]) ~]
+        ==
+      ::
+          %new-groupchat
+        :_  state
+        :~  :-  (welp here.bowl ~[%groupchats p/our.bowl t/name.poke])
+            [%make %groupchat ~ ~]
+        ==
+      ::
+          %invite-to-groupchat
+        =/  provider  ~[p/ship.poke %home %messenger]
+        :_  state
+        :~  :-  (welp here.bowl ~[%groupchats p/our.bowl t/name.poke])
+            [%poke groupchat-diff/!>([%invite ship.poke provider])]
+        ==
+      ==
+    ==
+  --
+--
+```
+
+## The +deps arm
+The only part of this system that uses the `+deps` arm is `/message-sub`.
+
+```hoon
+::
+::  define dependencies
+++  deps
+  ::  deps:neo is (map term fief:neo)
+  %-  ~(gas by *deps:neo)
+  :~  :*  %pub
+          ::  fief:neo is [required=? =quay]
+          req=&
+          ::  quay:neo is (pair lash:neo (unit port:neo)
+          ::  lash:neo defines the shrub's state and pokes
+          [[%pro %sig] (sy %sig ~)]
+          %-  some
+          ::  port:neo constrains the shrub's kids
+          :-  %y
+          %-  ~(gas by *lads:neo)
+          :~  :-  [[%.n %da] |]
+              [[%pro %message] ~]
+          ==
+      ==
+  ==
+```
+
+With regards to the lifecycle of the shrub, the `+deps` arm types the shrubs whose names and locations are passed in as an argument in the `%make` card when this shrub is created.
+
+```hoon
+[%make %message-sub ~ (malt ~[[%pub (snoc host.poke %pub)]])]
+```
+
+In this card, `+malt` creates a `(map term pith:neo)` where the `term` is `%pub` and the path represented by the `pith` is `/path/to/chat/pub`. At `/path/to/chat/pub`, there’s a shrub which is the canonical “publisher” or “host” of a chat, whether that’s a group chat or a DM. Whenever there’s a state change in the publisher shrub, the `/message-sub` created in this card will be notified about it, but we’ll cover that in more detail in the next section.
```

Comment:
Don't introduce the concept of chat just yet, because that is a higher-layer abstraction. So rather than saying "whether that’s a group chat or a DM" with those specific examples, we instead want to drive in the genericity of message-sub in this section, and then show how that genericity applies to the higher-level use cases